### PR TITLE
pca fix progress bar

### PR DESCRIPTION
**Pull request by Moo-Marc**
Improved progress bar "nesting" for recent modifications (PCA, flattening, scouts)

### CODE REVIEW

File: toolbox/math/bst_pca.m [hunk 1]
```diff
@@ -663,6 +667,14 @@
         end
 end
 
+if isProgressBar
+    % Reset existing progress bar as best we can; no way to know previous title or text.
```
Comment (rcassani) [inline toolbox/math/bst_pca.m:671]: Previous Title and Text could be obtained from the swing objects:
```
oldValue = bst_progress('get');
% Set is used to retrieve the bar structure
oldBar   = bst_progress('set', oldValue);            
oldTitle = oldBar.jWindow.getTitle().toCharArray';
oldText  = oldBar.jLabel.getTitle().toCharArray';
```

**Comment by Edouard2laire** [comment]
This is interesting. I am wondering if instead of 'nesting' progress bar, if we should not instead allow to have multiple progress bars (#614)

Maybe something like this (https://www.mathworks.com/matlabcentral/fileexchange/26589-multiwaitbar):

**Comment by Moo-Marc** [comment]
Thanks @rcassani, I'll make that a new bst_progress command: pBar = bst_progress('reset', pBar)
Would you have another suggestion instead of calling it 'reset'?  Maybe 'restart', or even overload 'start'?

Hi @Edouard2laire, indeed I saw your post at the time and like it, but I doubt we'll have resources for such improvements soon, with François being away.  This PR is just a quick fix to follow Bst "nesting conventions" - my recent changes had made some processes stick at 100% (e.g. connectivity with PCA scouts).

**Comment by rcassani** [comment]
Thanks @Edouard2laire, indeed that PR is in the queue of things to do.

> Would you have another suggestion instead of calling it 'reset'? Maybe 'restart', or even overload 'start'?

I think we can add two methods 'GetBar' and 'SetBar'. 
Will do

```
oldBar = bst_progress('GetBar');
bst_progress('SetBar', oldBar);

```

File: toolbox/math/bst_pca.m (hunk 1 above)
Comment (rcassani) [reply toolbox/math/bst_pca.m]: Added in [05021b2](https://github.com/brainstorm-tools/brainstorm3/pull/626/commits/05021b2c73a2f9f66b77667f1879231ceb7dfc81)

**Comment by rcassani** [comment]
@Moo-Marc, last two commits should do the trick. Can you test that it behaves as expected?

**Comment by Moo-Marc** [comment]
@rcassani  Few additional tweaks.  Works ok now!

**Pull request merged**